Sources/ProjectDescription/Dependencies/SwiftPackageManagerDependencies.swift

```diff
@@ -5,10 +5,15 @@ public struct SwiftPackageManagerDependencies: Codable, Equatable {
     /// List of packages that will be installed using Swift Package Manager.
     public let packages: [Package]
 
+    /// The `Product` type to be used for SPM targets with `automatic` library type.
+    public let automaticProductType: Product
+
     /// Creates `SwiftPackageManagerDependencies` instance.
     /// - Parameter packages: List of packages that will be installed using Swift Package Manager.
-    public init(_ packages: [Package]) {
+    /// - Parameter productType: The `Product` type to be used for SPM targets with `automatic` library type.
+    public init(_ packages: [Package], automaticProductType: Product = .staticLibrary) {
```

Comment:
I'm not sure if this should be a "global" switch - I could imagine people would want to configure this on a package-by-package basis. Plus if we did it this way, you wouldn't be able to do what SPM automatic linking does (that is link dynamically only libraries that are necessary).

Reply:
Makes sense, I'll add it to the `Package` then 🙌

Reply:
the problem is that we will have also to handle packages and targett not defined here (e.g. dependencies).
I'll replace the `automaticProductType: Product` here with a `customProductType: [String: Product]` which will allow to map any target to a custom product if needed.

Makes sense?

Reply:
ahh, right, I guess that's the most bulletproof approach. cc @ollieatkinson @laxmorek for possible additional thoughts (especially @ollieatkinson has already done some research regarding the automatic linking)

Reply:
Changed here: https://github.com/tuist/tuist/pull/3105/commits/6eeccc63e6e409cbe3a082174fdb5e891ad7fad9

Reply:
Being able to automatically determine the linking type is something Tuist should offer through Dependencies.swift, however, I like what is done here as a way to specify an override behaviour when the automatic choice does not meet your requirement. If this is purely additive and not required to specify for each package dependency then the change looks good 👍 Thanks @danyf90

Reply:
Ideally yes, but we would need a way here (i.e. in `ProjectDescription.TargetDependency` and `TuistGraph.TargetDependency`) to specify something like `automaticFramework` and then should be solved later on during graph generation, as the information is not available yet at this point of the pipeline.

I think that defaulting to staticFramework and allowing to override it in the `Dependencies.swift` should be enough for now. I have opened [an issue](https://github.com/tuist/tuist/projects/6#card-64009985) to keep track of the limitation

Reply:
sounds good @danyf90 👍